### DISCUSSION TITLE
Remove `Mesh2' from grid dimensions and variables

### DISCUSSION THREAD

**Discussion by hetland** [UXARRAY/uxarray#387] [General]
I would suggest to remove `Mesh2' from the grid dimensions and variables. This is used in a few example of 2D meshes in the ugrid conventions, but these are clearly just examples in the netCDF header, the associated tables showing the variables are probably more appropriate for naming. I am concerned this will be confusing to users, and will make working with uxarray more difficult. I understand this would touch quite a bit of the code, but I think better to shift to a better representation of the conventions now rather than later.

**Comment by rajeeja** [comment]
True, if we make a change sooner the better. What do you suggest instead? UXarray_node_x or uxmesh_node_x?

**Comment by rljacob** [comment]
I agree with this and like "uxmesh" instead of "Mesh2".

**Comment by philipc2** [comment]
@hetland 

This is a great point, and is something I've considered in the past. 

> What do you suggest instead? UXarray_node_x or uxmesh_node_x?

I don't see why we even need a header (`Mesh2`, `UxMesh`, etc) in the first place, since in my opinion it doesn't provide any relevant information. Right now our `Grid` is following the "**2D flexible mesh (mixed triangles, quadrilaterals, etc.) topology**" as outlined in the UGRID conventions, and until we plan to support other's (Triangular, 1D, 3D stacked, etc), we shouldn't need to worry about the heading since our `Grid` topology is always known.

Related but not exactly relevant to this discussion, but if we do chose to support other types of unstructured grid topologies, let's say 3D stacked, I would make much more sense to have a separate `Grid` for each topology (`Grid` would become `Grid2D` and 3D would be something like `Grid3D`), but this is a topic for much later.

We could keep the `UGRID` style simply by dropping the `Mesh2` and doing something like this:

#### Dimensions
`n_node`
`n_edge`
`n_face`
`n_max_face_nodes`
`n_nodes_per_face`

#### Coordinates
`node_x`: Default `x` Coordinate, in Degrees (-180, 180)
`node_y`: Default `y` Coordinate, in Degrees (-90, 90)
`node_x_cartesian` 
`node_y_cartesian` 
`node_z_cartesian` 


#### Connectivity
`face_nodes` or `face_nodes_connectivity`
_Same would be applied to all other connectivity variables, I prefer the shorter name but the longer one is more clear._

**Reply by hetland** [reply]
@philipc2 +1

I also think no specific header is needed. You can add a name attribute but really the grid instance is the name. Your suggestions are in line with the ugrid conventions.

**Comment by rajeeja** [comment]
+1 we're missing GooglePlus here :P

Let's drop the prefix Mesh2. @erogluorhan what do you think?

**Reply by erogluorhan** [reply]
This is great discussion, and I agree with @philipc2 's suggestion above. Hey @hetland thanks very much for the suggestion, and @hrajagers thanks very much for your feedback!

**Reply by erogluorhan** [reply]
I suggest we create an issue out of this discussion and start to work on it. @rajeeja @philipc2

**Reply by philipc2** [reply]
#389 & #396 has already begun this, will revisit it this week.

**Comment by hetland** [comment]
I've got a few minor points to add

> Right now our `Grid` is following the "**2D flexible mesh (mixed triangles, quadrilaterals, etc.) topology**" as outlined in the UGRID conventions, and until we plan to support other's (Triangular, 1D, 3D stacked, etc), we shouldn't need to worry about the heading since our `Grid` topology is always known.

I think keeping the scope of `uxarray` to 2D flexible meshes for the foreseeable future is very reasonable. This is in line with `xarray`'s focus on geophysical analysis, and basically all earth system models have a separate (separable) representation of the vertical coordinate -- I don't know of any regional or global ESMs that use a fully 3D unstructured mesh.

> #### Coordinates
> `node_x`: Default `x` Coordinate, in Degrees (-180, 180) `node_y`: Default `y` Coordinate, in Degrees (-90, 90) `node_x_cartesian` `node_y_cartesian` `node_z_cartesian`

To be general and clear to everyone, these should be called:
    `node_lon`: Default `x` Coordinate, in Degrees (-180, 180) 
    `node_lat`: Default `y` Coordinate, in Degrees (-90, 90) 
    `node_x` 
    `node_y` 
    `node_z`

**Reply by philipc2** [reply]
> To be general and clear to everyone, these should be called:
> node_lon: Default x Coordinate, in Degrees (-180, 180)
> node_lat: Default y Coordinate, in Degrees (-90, 90)
> node_x
> node_y
> node_z

This makes sense and is a good idea!

**Comment by hrajagers** [comment]
Dropping the `Mesh2` prefix makes total sense if there is only one mesh in the file. I introduced the prefix in the UGRID examples mainly to work towards the case of a mesh mosaic composed multiple parts. That's a feature (described [here](https://publicwiki.deltares.nl/display/NETCDF/Mosaics+of+meshes)) that we envisioned for (and have partially adapted in) our Delft3D FM modelling system, but that didn't make it into the official UGRID 1.0 standard.

**Reply by hetland** [reply]
@hrajagers - I agree that keeping grids distinct within a file is essential. But in this case, Mesh2_* is hardwired into the code -- this is not like a namespace that can be changed. The attribute prefixes will always be Mesh2. 

Second, other codes use different grid files for different nested grids (e.g., ROMS does this), and I would propose that it makes sense for each uxarray grid instance to be a separate, distinct grid. A group of nested grids could be put into a structure outside of uxarray, maybe even just a list or dictionary would be sufficient, if needed.
